src/libs/actions/Welcome.ts

```diff
@@ -57,9 +136,8 @@ Onyx.connect({
     key: ONYXKEYS.NVP_INTRO_SELECTED,
     initWithStoredValues: true,
     callback: (value) => {
-        hasSelectedChoice = !!value;
-
-        checkOnReady();
+        hasSelectedPurpose = !!value;
+        isAbleToDetermineOnboardingStatus({onAble: resolveOnboardingFlowStatus});
```

Comment:
No - all is good.`ONYXKEYS.NVP_INTRO_SELECTED` calls `isAbleToDetermineOnboardingStatus` instead of `checkOnReady`
```
Onyx.connect({
    key: ONYXKEYS.NVP_INTRO_SELECTED,
    initWithStoredValues: true,
    callback: (value) => {
        hasSelectedPurpose = !!value;
        isAbleToDetermineOnboardingStatus({onAble: resolveOnboardingFlowStatus});
    },
});
```